lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -423,6 +423,14 @@ export class SourceSelector extends React.Component<
 						// this will send an event down the ipcMain asking for metadata
 						// we'll get the response through an event
 
+						// FIXME: This is a poor man wait while loading to prevent a potential race condition without completely blocking the interface
+						// This should be addressed when refactoring the GUI
+						let retryRequestMetada = 10;
```

Comment:
```suggestion
						let retryRequestMetadata = 10;
```

Or even better, just “retriesLeft” or similar.